docs/genai/deployment/genai_serving_graph.md

```diff
@@ -0,0 +1,109 @@
+(genai-serving-graph)=
+# Gen AI realtime serving graph
+
+During inference, it is common to serve a gen AI model as part of a larger pipeline that includes data preprocessing, model execution, and post-processing. This can be done with MLRun using the real-time serving pipeline feature. Prior to model inference, the context is typically enriched using a vector database, then the input is transformed to input tokens, and finally the model is executed. Pre-processing and post-processing may also include guardrails to ensure the input is valid (for example, prevent the user from asking questions that attempt to exploit the model) as well as output processing, to verify the model does not hallucinate or include data that must not be shared.
+
+## A basic graph
+
+To run a model as part of a larger pipeline, you can use the `set_topology` method of the serving function. The following code shows how to set up a simple pipeline that includes a single step. This example is taken from the [Interactive bot demo using LLMs and MLRun](https://github.com/mlrun/demo-llm-bot) which calls OpenAI ChatGPT model:
```

Comment:
Maybe point to the `set_topology` documentation? Also, `set_topology` is only used in the next code snip, after defining the step. Maybe move the "you can use the `set_topology` method of the serving function" note there?